src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -260,12 +296,22 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
             }
 
             _functions = functions ?? new List<FunctionMetadata>();
-            _processStartupInterval = workerConfig.CountOptions.ProcessStartupInterval;
-            _restartWait = workerConfig.CountOptions.ProcessRestartInterval;
-            _shutdownTimeout = workerConfig.CountOptions.ProcessShutdownTimeout;
-            ErrorEventsThreshold = 3 * (await _maxProcessCount.Value);
 
-            if (Utility.IsSupportedRuntime(_workerRuntime, _workerConfigs))
+            if (_environment.IsMultiLanguageRuntimeEnvironment())
+            {
+                _processStartupInterval = _workerConfigs.Max(wc => wc.CountOptions.ProcessStartupInterval);
+                _restartWait = _workerConfigs.Max(wc => wc.CountOptions.ProcessRestartInterval);
+                _shutdownTimeout = _workerConfigs.Max(wc => wc.CountOptions.ProcessShutdownTimeout);
+            }
+            else
+            {
+                _processStartupInterval = workerConfig.CountOptions.ProcessStartupInterval;
+                _restartWait = workerConfig.CountOptions.ProcessRestartInterval;
+                _shutdownTimeout = workerConfig.CountOptions.ProcessShutdownTimeout;
+            }
+            ErrorEventsThreshold = 3 * await _maxProcessCount.Value;
```

Comment:
Avoid hardcoded magical values in code. Consider defining a named constant.

Reply:
Yeah, this has been around for a long time -- don't worry about updating it here (just to keep the changes minimal).